### ISSUE TITLE
Feature request: Add setting on server-side to restrict group listing

### ISSUE THREAD

**Issue by ZetaWaves** [rustdesk/rustdesk-server-pro#250]
Please add a setting on server-side to restrict accessing the group list to logged-in users.

We want to allow users to log in and connect to anyone they want, but we don't want to give them a list of target machines. We want them to have to know the ID of the machine to connect. Giving them a list of machines, even though it's still secure, can create a problem. Please add this setting to add a bit of security by obscurity. 

Basically, the same setting as "only admin can access unassigned devices", except that all users can connect to anyone, they just can't see a list of all the ID's.

I am proposing either:
1.) Add the feature as a server-side option.
-or-
2.) Add a checkbox to custom client creation to disable the group panel from the client altogether so it doesn't show up as an option at all.

**Comment by ZetaWaves** [comment]
linked to #244

**Comment by Streiter00** [comment]
Same here. That s very important for me too!
I tried to work around ad add them to different user to get them all out of the default but the you have to name every device manually by hand whats much to complex working if your devices gone higher

**Comment by rustdesk** [comment]
ok, we add to custom client

**Comment by rustdesk** [comment]
> Same here. That s very important for me too!
> I tried to work around ad add them to different user to get them all out of the default but the you have to name every device manually by hand whats much to complex working if your devices gone higher

Is this related to hiding group list?

**Comment by Streiter00** [comment]
> > Same here. That s very important for me too!
> > I tried to work around ad add them to different user to get them all out of the default but the you have to name every device manually by hand whats much to complex working if your devices gone higher
> 
> Is this related to hiding group list?

Yes because i tried to hide the default group list by adding the default devices to a user to get the default list empty

**Comment by rustdesk** [comment]
https://x.com/rustdesk/status/1796114741116469643
